src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/GeneratorDiagnostic.cs

```diff
@@ -55,8 +57,130 @@ public override DiagnosticInfo ToDiagnosticInfo(DiagnosticDescriptor descriptor,
                     UnnecessaryDataLocations,
                     // Add "unnecessary locations" property so the IDE fades the right locations.
                     DiagnosticProperties.Add(WellKnownDiagnosticTags.Unnecessary, $"[{string.Join(",", Enumerable.Range(0, UnnecessaryDataLocations.Length))}]"),
-                    UnnecessaryDataDetails,
-                    elementName);
+                    UnnecessaryDataName,
+                    elementName,
+                    UnnecessaryDataDetails ?? "");
+            }
+        }
+
+        /// <summary>
+        /// Provides an implementation of <see cref="IMarshallingGenerator.SupportsByValueMarshalKind(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/> through <see cref="GetSupport(ByValueContentsMarshalKind, TypePositionInfo, StubCodeContext, out GeneratorDiagnostic?)"/>
+        /// </summary>
+        public record ByValueMarshalKindSupportManager(
+            ByValueMarshalKindSupport InSupport, string? InSupportDetails,
+            ByValueMarshalKindSupport OutSupport, string? OutSupportDetails,
+            ByValueMarshalKindSupport InOutSupport, string? InOutSupportDetails)
+        {
+            /// <summary>
+            /// A default <see cref="ByValueMarshalKindSupportManager"/> for by value parameters. [In] is allowed, but unnecessary. Out is not allowed.
+            /// </summary>
+            public static ByValueMarshalKindSupportManager ValueTypeParameterDefault = new ByValueMarshalKindSupportManager(
```

Comment:
Is this for value type parameters or for parameters passed by value? If it's for "by value" parameters, can we rename this to `ByValueParameterDefault`?

Reply:
This is for value types.